StackMin.py

```diff
@@ -0,0 +1,33 @@
+class Stack:
+    def __init__(self):
+        self.dataStack = []  
+        self.minStack = []   
+
+    def push(self, val):
+        self.dataStack.append(val)
+        if not self.minStack or val <= self.minStack[-1]:
+            self.minStack.append(val)
+
+    def pop(self):
+        if self.dataStack:
+            if self.dataStack[-1] == self.minStack[-1]:
+                self.minStack.pop()
+            return self.dataStack.pop()
+
+    def min(self):
+        if self.minStack:
+            return self.minStack[-1]
+        return None
```

Comment:
Because you're returning in both the `if` and (implied) `else` cases with no extra computation, you can do this cleanly in one line using a ternary.
```suggestion
        return self.minStack[-1]  if self.minStack else None
```